src/components/App/App.jsx

```diff
@@ -147,59 +196,87 @@ function App() {
-          <Routes>
-            <Route
-              path="/"
-              element={
-                <Main
-                  handleHamburgerClick={handleHamburgerClick}
-                  handleLoginClick={handleLoginClick}
-                  isLoggedIn={isLoggedIn}
-                  handleLogout={handleLogout}
-                  handleSearchSubmit={handleSearchSubmit}
-                  newsData={newsData}
-                  isSuccess={isSuccess}
-                  isLoading={isLoading}
-                  isError={isError}
-                  setCurrentKeyword={setCurrentKeyword}
-                />
-              }
-            ></Route>
-            <Route
-              path="/saved-news"
-              element={
-                <SavedNews
-                  handleHamburgerClick={handleHamburgerClick}
-                  isLoggedIn={isLoggedIn}
-                  currentUser={currentUser}
-                  handleLogout={handleLogout}
-                />
-              }
-            ></Route>
-          </Routes>
-          <Footer />
-          <LoginModal
-            isOpen={activeModal === "login"}
-            onClose={closeActiveModal}
-            setActiveModal={setActiveModal}
-            handleLogin={handleLogin}
-          />
-          <RegisterModal
-            isOpen={activeModal === "register"}
-            onClose={closeActiveModal}
-            setActiveModal={setActiveModal}
-            handleRegistration={handleRegistration}
-          />
-          <SuccessModal
-            isOpen={activeModal === "success"}
-            onClose={closeActiveModal}
-            setActiveModal={setActiveModal}
-          />
-        </div>
+        <UserArticleContext.Provider value={userArticleContext}>
+          <div className="app__content">
+            <Routes>
+              <Route
+                path="/"
+                element={
+                  <Main
+                    handleLoginClick={handleLoginClick}
+                    isLoggedIn={isLoggedIn}
+                    handleLogout={handleLogout}
+                    handleSearchSubmit={handleSearchSubmit}
+                    newsData={newsData}
+                    isSuccess={isSuccessNewsData}
+                    isLoading={isLoading}
+                    isError={isError}
+                    setCurrentKeyword={setCurrentKeyword}
+                    handleSaveArticle={handleSaveArticle}
+                    handleDeleteArticle={handleDeleteArticle}
+                  />
+                }
+              ></Route>
+              <Route
+                path="/saved-news"
+                element={
+                  <ProtectedRoute
+                    isLoggedIn={isLoggedIn}
+                    isAuthSettled={isAuthSettled}
+                    setActiveModal={setActiveModal}
+                  >
+                    <SavedNews
+                      isLoggedIn={isLoggedIn}
+                      currentUser={currentUser}
```

Comment:
`currentUser={currentUser}` is not needed in the props because it should be taken from `CurrentUserContext`